### PR TITLE
Added email functionality.

### PR DESCRIPTION
**Pull request by kadikraman**
Added an emailer.py which will handle sending emails.
Recipient addresses are specified in the config.
Sending via gmail and the gmail account address and password will be stored in credentials.json (formerly api_key.json)
I had to call the send email function in google_maps.py because doing it in housebot.by caused a race condition so the email was sent before all the data was gathered.

### CODE REVIEW

File: config.py [hunk 1]
```diff
@@ -34,3 +34,6 @@
   'Alice': '221b Baker Street, London NW1 6XE',
   'Bob': 'Buckingham Palace, London SW1A 1AA'
 }
+
+house_member_email_addresses = {
```
Comment (MarcelCutts) [inline config.py]: It would be great if there was an example in place for people to use and adapt, as with Alice and Bob's address.

Comment (MarcelCutts) [inline config.py]: Line note?

File: emailer.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    """ % (FROM, ", ".join(TO), SUBJECT, TEXT)
+    try:
+        server = smtplib.SMTP("smtp.gmail.com", 587)  # or port 465 doesn't seem to work!
+        server.ehlo()
+        server.starttls()
+        server.login(gmail_user, gmail_pwd)
+        server.sendmail(FROM, TO, message)
+        server.close()
+        print 'successfully sent the mail'
+    except:
```
Comment (MarcelCutts) [inline emailer.py]: This is too broad, I had to alter this to figure out what exception I was getting to get partway to fixing my issues in running this script. I would suggest maybe beginning with catching SMTPAuthenticationError?

File: readme.md [hunk 3]
```diff
@@ -3,15 +3,25 @@
...
 
-- investigate config.py and update it with housing preferences relevant for you
+- investigate config.py and update it with housing preferences relevant for you (add emails of all the people who
+want to be notified when new houses are found in the house_member_email_addresses in JSON format, i.e
+```
+{
+    "email": "email@gmail.com",
+    "email2": "email2@gmail.com"
+}
+```
```
Comment (MarcelCutts) [inline readme.md]: The above is a bit odd, why use a dict? Would a simpler list of emails make more sense?

File: config.py (hunk 1 above)
Comment (kadikraman) [inline config.py]: I thought of that, but I don't want to use an email address that actually exists! I guess I can just do example@gmail.com unless you have any other suggestions?

File: readme.md (hunk 3 above)
Comment (kadikraman) [inline readme.md]: The idea behind this was that you could keep track of whose email is which and maybe send emails with "Dear [name]". But I'm happy to change it to a list if that makes more sense.

Comment (MarcelCutts) [inline readme.md]: OK, I can get on board with that, even if you aren't using it now. Could it be smarter to combine the commuter addresses object with the email one, then? E.g.

```
house_member_details = [
{ "name": "Alice",
  "commute_destination": "221B Baker Street",
  "email_address": "alice@useagmailaddress.com",
  "max_commute_time_in_minutes": 45
}, 
  "name": "Bob",
  "commute_destination": "221B Baker Street",
  "email_address": "bob@useagmailaddress.com",
  "max_commute_time_in_minutes": 30
}]
```

File: config.py (hunk 1 above)
Comment (MarcelCutts) [inline config.py]: Sure, or a blatantly non-valid one, e.g. "your-address-at-gmail-dot-com"?

File: readme.md (hunk 3 above)
Comment (kadikraman) [inline readme.md]: I like this idea! Less places to edit stuff.

**Comment by MarcelCutts** [comment]
Tested locally, it works. Also implemented most of my suggestions, so I am appeased!

**Pull request merged**